parmed/amber/amberformat.py

```diff
@@ -490,36 +497,43 @@ def rdparm_slow(self, fname):
         fmtre = re.compile(r'%FORMAT *\((.+)\)')
         version = None
 
+        if isinstance(fname, string_types):
+            prm = genopen(fname, 'r')
+            own_handle = True
+        else:
+            prm = fname
+            own_handle = True
+            assert hasattr(prm, 'read'), 'must be a file object'
```

Comment:
Only use an `assert` for code that by design should only be hit in the case of a bug in *our* code.  Python has 2 modes of "optimization" when compiling:

1. `-O` == get rid of all asserts
2. `-OO` == get rid of all docstrings

(At least this is what I recall.)  The whole purpose here is to make `assert` less unpleasant to sprinkle around the code, since a simple flag will eliminate any possible performance hit in a production distribution.

This isn't particularly well-publicized though, which is why people tend to abuse it :)